rskj-core/src/main/java/co/rsk/peg/pegininstructions/PeginInstructionsProvider.java

```diff
@@ -0,0 +1,92 @@
+package co.rsk.peg.pegininstructions;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.script.ScriptChunk;
+import co.rsk.bitcoinj.script.ScriptOpCodes;
+import java.util.List;
+import java.util.Optional;
+import org.bouncycastle.util.encoders.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PeginInstructionsProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(PeginInstructionsProvider.class);
+
+    public Optional<PeginInstructions> buildPeginInstructions(BtcTransaction btcTx) throws
+        PeginInstructionsException {
+
+        logger.trace("[buildPeginInstructions] Using btc tx {}", btcTx.getHash());
+
+        PeginInstructions peginInstructions;
+        byte[] opReturnOutputData;
+
+        try {
+            opReturnOutputData = extractOpReturnData(btcTx);
+        } catch (Exception e) {
```

Comment:
Careful here. We must not treat every exception equally.
If there is no OP_RETURN we should return Optional.empty.
But if there is any other situation (more than one op_return, null data after op_return, or any unhandled exception) we should raise a new exception for the funds to be returned.